walkbox.go

```diff
@@ -239,25 +247,68 @@ func (wm *WalkBoxMatrix) FindPath(from, to Position) []*WayPoint {
 	var path []*WayPoint
 	fromf := from.ToPosf()
 	tof := to.ToPosf()
-	current, _ := wm.walkBoxAt(fromf)
-	target, _ := wm.walkBoxAt(tof)
+	current, _ := wm.walkBoxIndex(fromf)
+	target, _ := wm.walkBoxIndex(tof)
+	currentWp := &WayPoint{Walkbox: wm.walkBoxes[current], Position: from}
 
-	path = append(path, &WayPoint{Walkbox: wm.walkBoxes[current], Position: from})
+	path = append(path, currentWp)
 	for current != target {
 		next := wm.nextWalkBox(current, target)
 		if next == InvalidWalkBox {
 			break
 		}
 		nextPosition := wm.closestPositionToWalkBox(fromf, next)
-		path = append(path, &WayPoint{Walkbox: wm.walkBoxes[next], Position: nextPosition.ToPos()})
+		nextWp := &WayPoint{Walkbox: wm.walkBoxes[next], Position: nextPosition.ToPos()}
+		path = append(path, append(currentWp.interpolate(nextWp), nextWp)...)
 		current = next
+		currentWp = nextWp
 		fromf = nextPosition
 	}
 
 	path = append(path, &WayPoint{Walkbox: wm.walkBoxes[current], Position: wm.closestPositionOnWalkBox(current, tof).ToPos()})
 	return path
 }
 
+// interpolate generates intermediate waypoints between a starting and ending waypoint.
+// We need interpolate points if scale difference is higher.
+func (from *WayPoint) interpolate(to *WayPoint) []*WayPoint {
```

Comment:
I wouldn't make any interpolation. 

Interpolation is problematic since it will generate sprite frames of very odd dimensions. For example, if we use sprites whose dimensions are multiples of 8 (e.g., 32x48), we know we can multiply the sprite by any fraction of 8 (1/8, 2/8, 4/8, 6/8) and the number of pixels will always be integer. However, if we use interpolation, that will not occur and the number of pixels will be totally odd. As result, the image will look deformed. 

I think the right way to handle this is to have as many walkboxes as needed to have a smooth transition as the actor moves to the back of the room.